### PR TITLE
feat: [CDS-82613] Fix Schema For Traffic Routing

### PR DESCRIPTION
**Pull request by alebuh**

### CODE REVIEW

File: v0/pipeline/steps/cd/k8s-traffic-routing-provider-base-spec.yaml [hunk 1]
```diff
@@ -2,8 +2,6 @@ title: K8sTrafficRoutingProviderBaseSpec
 type: object
 allOf:
   - "$ref": "k8s-traffic-routing-destinations.yaml"
-required:
```
Comment (markoharness) [inline v0/pipeline/steps/cd/k8s-traffic-routing-provider-base-spec.yaml:5]: I am not getting this change. We should have routes with at least 1 item in array.

File: v1/pipeline/steps/cd/k8s-traffic-routing-provider-base-spec.yaml [hunk 2]
```diff
@@ -2,8 +2,6 @@ title: K8sTrafficRoutingProviderBaseSpec
 type: object
 allOf:
   - "$ref": "k8s-traffic-routing-destinations.yaml"
-required:
```
Comment (markoharness) [inline v1/pipeline/steps/cd/k8s-traffic-routing-provider-base-spec.yaml:5]: Same here

File: v0/pipeline/steps/cd/k8s-traffic-routing-provider-base-spec.yaml (hunk 1 above)
Comment (alebuh) [reply v0/pipeline/steps/cd/k8s-traffic-routing-provider-base-spec.yaml:5]: for SMI provider it's not required

File: v1/pipeline/steps/cd/k8s-traffic-routing-provider-base-spec.yaml (hunk 2 above)
Comment (alebuh) [reply v1/pipeline/steps/cd/k8s-traffic-routing-provider-base-spec.yaml:5]: for SMI provider it's not required

**Review by markoharness — APPROVED** [review]

**Review by ivan-83 — APPROVED** [review]

**Review by acasian-harness — APPROVED** [review]

**Pull request merged**